datafusion/physical-expr/src/aggregate/covariance.rs

```diff
@@ -291,19 +291,20 @@ impl Accumulator for CovarianceAccumulator {
         let mut arr1 = downcast_value!(values1, Float64Array).iter().flatten();
         let mut arr2 = downcast_value!(values2, Float64Array).iter().flatten();
 
-        for _i in 0..values1.len() {
-            let value1 = arr1.next();
-            let value2 = arr2.next();
+        for i in 0..values1.len() {
+            let value1 = match values1.is_valid(i) {
+                true => arr1.next(),
+                false => None,
+            };
+            let value2 = match values2.is_valid(i) {
+                true => arr2.next(),
+                false => None,
+            };
 
             if value1.is_none() || value2.is_none() {
```

Comment:
Similar to the comment above, if we had Rust 1.65 I would use the `let-else` construct to get rid of the unwraps.

Reply:
Argh - I hate GH's line selection. I am referring to the `continue` check right below.